trunk/source/CACSL2BoogieTranslator/src/de/uni_freiburg/informatik/ultimate/cdt/translation/implementation/base/standardfunctions/StandardFunctionHandler.java

```diff
@@ -1832,6 +1839,55 @@ private Result handleAssert(final IDispatcher main, final IASTFunctionCallExpres
 				name, mSettings.checkAssertions(), Spec.ASSERT, result.getLrValue().getValue())).build();
 	}
 
+	/**
+	 * Handle C11 or C23 static assertions with or without an explicit message.
+	 * 
+	 * @param main
+	 *            the current dispatcher
+	 * @param node
+	 *            the static assert expression
+	 * @param loc
+	 *            the location of the static assert
+	 * @param name
+	 *            the name of the method
+	 * 
+	 * @return {@link ExpressionResult} representing the static assertion
+	 */
+	private Result handleStaticAssert(final IDispatcher main, final IASTFunctionCallExpression node,
+			final ILocation loc, final String name) {
+
+		final IASTInitializerClause[] arguments = node.getArguments();
+		final int numAssertArgs = arguments.length;
+
+		/* check if signature of assertion is of form 'static_assert(expr)' or 'static_assert(expr, msg)' */
+		if (numAssertArgs == 2) {
+			/* static C11 or C23 assertion with two arguments (expr and msg) */
+			checkArguments(loc, 2, name, arguments);
+
+			if (isStringLiteral(arguments[1])) {
+				/* extract string literal value for custom error message */
+				final String errorMsg = String.valueOf(((IASTLiteralExpression) arguments[1]).getValue());
+
+				final ExpressionResult result = mExprResultTransformer
+						.transformSwitchRexIntToBool((ExpressionResult) main.dispatch(arguments[0]), loc, node);
+				return new ExpressionResultBuilder()
+						.addAllExceptLrValue(result).addStatement(createReachabilityAssert(loc, name,
+								mSettings.checkAssertions(), Spec.ASSERT, result.getLrValue().getValue(), errorMsg))
+						.build();
+			} else {
+				/* WARNING: this case should be never reached since the msg should be always a string literal */
+				/* TODO: Fallback: handle as regular assertion without any msg */
+				throw new IncorrectSyntaxException(loc, "Message parameter of static assert is not a string literal");

```

Comment:
Not really a problem for now, but you could handle this quite easily, right? Just dispatch `arguments[1]` and create an `ExpressionResult` with that and `handleAssert(main, node, loc, name)`.

Reply:
You're right! This would be an easy solution for the proposed TODO.

At the moment, the intended behavior from the C11 and C23 standard is implemented. The proposed TODO is only one possibility to weaken this exceptional case, but is not covered by any C standard. So it is up to you: Should we weaken this exceptional case or not?

Reply:
Ah, okay, if there is only a string literal allowed by the C standard, then `IncorrectSyntaxException` is fine here. I just misread your comment here, so you could please update it and state that this case is not allowed (instead of the TODO)?

Reply:
I will remove the TODO to be compliant with the C standard.